### PR TITLE
Allow scalars to accept Neo4j native values

### PR DESCRIPTION
**Pull request by darrellwarde**
# Description

> **Note**
> 
> Please provide a description of the work completed in this PR below

Allows scalars to accept Neo4j types rather than strings only for temporal types - better compatibility with OGM.

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High

Complexity: Low

# Issue

> **Note**
>
>  Please link to the GitHub issue(s) in which the proposal for this work was discussed
>  
>  To link to multiple issues, use full syntax for each, for example `Closes #1, closes #2, closes #3`

Supersedes #1099

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-2503.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/ogm/tests/integration/scalars.int.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * limitations under the License.
+ */
+
+import { Driver, Session, Date as Neo4jDate, DateTime, Duration, LocalDateTime, LocalTime, Time } from "neo4j-driver";
+import { gql } from "graphql-tag";
+import neo4j from "./neo4j";
+import { OGM } from "../../src";
+import { generateUniqueType } from "../utils";
+
+describe("pluralize with underscore", () => {
```
Comment (mjfwebb) [inline packages/ogm/tests/integration/scalars.int.test.ts:26]: Description needs to be updated :)

**Review by mjfwebb — CHANGES_REQUESTED** [review]

File: packages/ogm/tests/integration/scalars.int.test.ts (hunk 1 above)
Comment (darrellwarde) [reply packages/ogm/tests/integration/scalars.int.test.ts]: Good spot, thanks!

**Review by mjfwebb — APPROVED** [review]
Looks great! :+1:

**Pull request merged**